### PR TITLE
MM-29858: Migrate string refs to functional ones in autosize textarea component

### PR DESCRIPTION
**Pull request by sowmiyamuthuraman**
Fixes https://github.com/mattermost/mattermost-server/issues/16041

JIRA ticket: https://mattermost.atlassian.net/browse/MM-29858

### CODE REVIEW

File: components/autosize_textarea.tsx [hunk 1]
```diff
@@ -17,42 +16,53 @@ type Props = {
...
     private height: number;
+    private textareaRef: React.RefObject<HTMLTextAreaElement>;
+    private reference: React.RefObject<HTMLTextAreaElement>;
     constructor(props: Props) {
         super(props);
 
         this.height = 0;
+
+        this.textareaRef = React.createRef();
+        this.reference = React.createRef();
```
Comment (nevyangelova) [inline components/autosize_textarea.tsx:27]: nit: can we keep the convention of naming this `this.referenceRef`

Comment (sowmiyamuthuraman) [reply components/autosize_textarea.tsx:27]: Since the instance varibale name is already reference thought of not adding it initially.  Will fix this

**Comment by jfrerich** [comment]
@Willyfrog, @nevyangelova - kind reminder to do reviews

**Review by nevyangelova — APPROVED** [review]
Thanks! 🎉

File: components/autosize_textarea.tsx [hunk 2]
```diff
@@ -17,42 +16,53 @@ type Props = {
 
 export default class AutosizeTextarea extends React.PureComponent<Props> {
     private height: number;
+    private textareaRef: React.RefObject<HTMLTextAreaElement>;
```
Comment (Willyfrog) [inline components/autosize_textarea.tsx:19]: minor nit
```suggestion
    private textAreaRef: React.RefObject<HTMLTextAreaElement>;
```
we try to do camelCase throughout the code, so in order to be consintent we should rename it.

**Review by Willyfrog — CHANGES_REQUESTED** [review]
Thanks for the PR. It looks good, but there is a really small nit about conventions, can you fix it?

**Review by Willyfrog — APPROVED** [review]
Thank you for the contribution!! looking good to me!

**Comment by jgilliam17** [comment]
Thank you @sowmiyamuthuraman 
I am seeing an issue on the test server - text remains visible in the textbox after posting. Can you please take a look? 
Screen Shot 2020-11-04 at 2 35 58 PM
Screen Shot 2020-11-04 at 2 45 11 PM

**Comment by sowmiyamuthuraman** [comment]
> Thank you @sowmiyamuthuraman
> I am seeing an issue on the test server - text remains visible in the textbox after posting. Can you please take a look?
> Screen Shot 2020-11-04 at 2 35 58 PM
> Screen Shot 2020-11-04 at 2 45 11 PM

@jgilliam17 I have fixed this.. can you please check now?

**Review by jgilliam17 — APPROVED** [review]
Thank you @sowmiyamuthuraman 
Tested, looks good to merge.
- Verified textarea - as before, no change in functionality.
@Willyfrog  can you please help merge? Thanks 🙂

**Pull request merged**